### PR TITLE
fix: missing system message when an adding member to a conversation AR-2572

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

No system message is inserted when adding a user to a conversation.

### Causes

Not implemented in the `AddMemberToConversationUseCase`

### Solutions

Add the missing implementation and align adding a member with removing a member.

#### Test Coverage

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by MohamadJaara** [comment]
i'm 99.99% sure this has been already fixed by @MohammedMokresh but the fix is in the release branch 😅

**Comment by MohammedMokresh** [comment]
> i'm 99.99% sure this has been already fixed by @MohammedMokresh but the fix is in the release branch 😅

https://github.com/wireapp/kalium/pull/989

**Comment by typfel** [comment]
😐 I don't like this modus operandi with the release branch.

But anyway I fixed some structural issues with MLS and tests so I'd like to update this PR when develop has all the changes from the release branch.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt [hunk 1]
```diff
@@ -494,39 +493,42 @@ internal class ConversationDataSource internal constructor(
...
+            val users = userIdList.map {
+                idMapper.toApiModel(it)
+            }
+            val addParticipantRequest = AddConversationMembersRequest(users, DEFAULT_MEMBER_ROLE)
+            conversationApi.addMember(
+                addParticipantRequest, idMapper.toApiModel(conversationId)
+            )
+        }.flatMap { response ->
+            val memberList = userIdList.map { userId ->
+                // TODO: mapping the user id list to members with a made up role is incorrect and a recipe for disaster
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:521]: don't we receive the member role here from the API? shouldn't we address the above todo here?

**Review by mchenani — APPROVED** [review]
LGTM🚀 + questions

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt (hunk 1 above)
Comment (typfel) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:521]: We should, but I'd argue not in this PR.

**Pull request merged**